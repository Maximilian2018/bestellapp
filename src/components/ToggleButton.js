import React, { Component } from 'react'
import styled from 'react-emotion'

const StyledButton = styled('div')`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 40px;
  background: ${props => (props.isSelected ? 'cornflowerblue' : 'white')};
`

export default class ToggleButton extends Component {
  render() {
    return (
      <StyledButton
        isSelected={this.props.isSelected}
        onClick={this.props.toggle}
      >
        {this.props.text}
      </StyledButton>
    )
  }
}
