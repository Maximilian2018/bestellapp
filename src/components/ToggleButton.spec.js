import React from 'react'
import ToggleButton from './ToggleButton'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('ToggleButton', () => {
  let wrapper
  let mockFunction = jest.fn()

  beforeEach(() => {
    wrapper = mount(
      <ToggleButton isSelected={false} toggle={mockFunction} text={'foo'} />
    )
  })

  it('renders a ToggleButton', () => {
    expect(wrapper.html()).toContain('foo')
  })

  it('reacts on a click', () => {
    expect(mockFunction.mock.calls.length).toBe(0)
    wrapper.simulate('click')
    expect(mockFunction.mock.calls.length).toBe(1)
  })

  it('changes the color', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.setProps({ isSelected: true })
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
