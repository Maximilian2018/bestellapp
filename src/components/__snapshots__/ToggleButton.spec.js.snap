// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton changes the color 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 40px;
  background: yellow;
}

<ToggleButton
  isSelected={false}
  text="foo"
  toggle={[Function]}
>
  <Styled(div)
    isSelected={false}
    onClick={[Function]}
  >
    <div
      className="emotion-0"
      onClick={[Function]}
    >
      foo
    </div>
  </Styled(div)>
</ToggleButton>
`;

exports[`ToggleButton changes the color 2`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 40px;
  background: cornflowerblue;
}

<ToggleButton
  isSelected={true}
  text="foo"
  toggle={[Function]}
>
  <Styled(div)
    isSelected={true}
    onClick={[Function]}
  >
    <div
      className="emotion-0"
      onClick={[Function]}
    >
      foo
    </div>
  </Styled(div)>
</ToggleButton>
`;
